Clarified action of function and added example (#25368)
mdn/content: files/en-us/web/api/event/stopimmediatepropagation/index.md

If several listeners are attached to the same element for the same event type, they are
called in the order in which they were added. If `stopImmediatePropagation()`
is invoked during one such call, no remaining listeners will be called, either on that
element or any other element.

## Syntax

```js-nolint
event.stopImmediatePropagation()
```

> ### Comparing event stopping functions

Reviewer: We have quite a particular structure for reference pages: https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Page_types/API_method_subpage_template.

Especially, examples should go inside their own "Examples" H2, and also see the formatting we use for live samples: https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Code_examples#traditional_live_samples.

> The example below has three buttons inside of three nested divs. Each button has two
> event listeners registered for click events, and each div has an event listener, also
> registered for click events. The top button allows normal event propagation while the
> middle button calls `stopPropagation()` in its first callback function and the bottom
> button calls `stopImmediatePropagation()` in its first callback function.

Reviewer: ```suggestion
The example below has three buttons inside three nested {{HTMLElement("div")}} elements. Each button has two
event listeners registered for click events, and each `<div>` has an event listener, also
registered for click events.

- The top button allows normal event propagation.
- The middle button calls `stopPropagation()` in its first callback function
- The bottom button calls `stopImmediatePropagation()` in its first callback function.
```

Reviewer: We don't really encourage hard line breaks for prose either, although I wouldn't reject a PR for it.

Reviewer 2: I think the only "rule" is that we stick to rigidly is no one likes line breaks inside macros or within HTML tags inside HTML tables. 

FWIW I'm prefer semantically useful line breaks to arbitrary fixed-length lines. I'm in the minority that I like these breaks on sentences, while most of the docs either break on paragraphs or at random points.

#### HTML

```html
  <h2>Click on the buttons</h2>
  <div id="div1">div1<br><div id="div2">div2<br><div id="div3">div3<br>
    <button id="allow">allow propagation</button><br>
    <button id="stop">stop propagation</button><br>

>     <button id="istop">immediate stop propagation</button>

Reviewer: nit, but `"immediate-stop"` would be better than `"istop"`

  </div></div></div>
  <div id="output"></div>

>   <script>

Reviewer: This should be in a separate `### JavaScript` section, per https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Code_examples#traditional_live_samples.

Reviewer: You can rely on the DOM being available.

    const outElem = document.getElementById("output");

    /* Clear the output */
    document.addEventListener("click", () => outElem.innerHTML = "", true);

    [1, 2, 3].forEach(value => {
      const id = "div" + value;

>       document.getElementById(id).addEventListener("click", () =>

Reviewer: I guess this is a matter of personal taste but I would far rather repeat the `addEventListener` calls here.

Author: I think I have a more agreeable alternative that is still DRY.

>           outElem.innerHTML += "click event processed on " + id + "<br>"

Reviewer: We should not write to `innerHTML`. In cases like this I usually make the output element a `<pre>` and set its `textContent` (e.g. https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/progress_event#live_example).

Reviewer: We also prefer template literals to `+` for concatenation: https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide/Code_style_guide/JavaScript#template_literals.

Author: I had used innerHTML due to the <br> nodes, but <pre> actually allows some simplification besides getting rid of the innerHTML.

Author: I had used innerHTML due to the `<br>` nodes, but `<pre>` actually allows some simplification besides getting rid of the innerHTML.

        evt => {
          outElem.innerHTML += "click event 1 processed on " +
              "\"stop propagation\" button<br>" +
              "&nbsp;&nbsp;calling stopPropagation()<br>";
          evt.stopPropagation();
        });
    stopElem.addEventListener("click", () =>
        outElem.innerHTML += "click event 2 processed on " +
            "\"stop propagation\" button <br>"
        );

    const istopElem = document.getElementById("istop");
    istopElem.addEventListener("click",
        evt => {
          outElem.innerHTML += "click event 1 processed on " +
              "\"immediate stop propagation\" button<br>" +
              "&nbsp;&nbsp;calling stopImmediatePropagation()<br>";
          evt.stopImmediatePropagation();
        });
    istopElem.addEventListener("click", () =>
        outElem.innerHTML += "click event 2 processed on " +
            "\"immediate stop propagation\" button <br>"
        );

  </script>
```

```css
    #div1, #div2, #div3 {
      display: inline-block;
      padding: 20px;
      border: 2px solid #000;
    }
    button {
      width: 100px;
      color: #008;
      padding: 5px;
      margin: 5px;
    }
```

#### Result

> The click callback functions display messages when they are called.

Reviewer: ```suggestion
The click event handlers display messages when they are called.
```

> It is seen that calling `stopPropagation()` allows the other callback
> function registered for clicks on the button to execute while calling
> `stopImmediatePropagation()` does not.

Reviewer: ```suggestion
You will see that calling `stopPropagation()` allows the other event handlers registered for clicks on the button to execute while calling `stopImmediatePropagation()` does not.
```

Author: I resolved all of the issues and made the example a bit prettier. I see the commit I made, and I tried to create a new pull request. However, I am questioning if the pull request exists. Is there a new pull request?

Reviewer: No (as you probably saw by now) your new changes get added to this pull request. Which is what we want :).

Sorry I've not had a chance to review this today but I will tomorrow.
